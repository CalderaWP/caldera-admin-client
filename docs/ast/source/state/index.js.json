{
  "type": "File",
  "start": 0,
  "end": 126,
  "loc": {
    "start": {
      "line": 1,
      "column": 0
    },
    "end": {
      "line": 7,
      "column": 2
    }
  },
  "program": {
    "type": "Program",
    "start": 0,
    "end": 126,
    "loc": {
      "start": {
        "line": 1,
        "column": 0
      },
      "end": {
        "line": 7,
        "column": 2
      }
    },
    "sourceType": "module",
    "body": [
      {
        "type": "ImportDeclaration",
        "start": 0,
        "end": 42,
        "loc": {
          "start": {
            "line": 1,
            "column": 0
          },
          "end": {
            "line": 1,
            "column": 42
          }
        },
        "specifiers": [
          {
            "type": "ImportSpecifier",
            "start": 8,
            "end": 19,
            "loc": {
              "start": {
                "line": 1,
                "column": 8
              },
              "end": {
                "line": 1,
                "column": 19
              }
            },
            "imported": {
              "type": "Identifier",
              "start": 8,
              "end": 19,
              "loc": {
                "start": {
                  "line": 1,
                  "column": 8
                },
                "end": {
                  "line": 1,
                  "column": 19
                },
                "identifierName": "dispatchers"
              },
              "name": "dispatchers"
            },
            "local": {
              "type": "Identifier",
              "start": 8,
              "end": 19,
              "loc": {
                "start": {
                  "line": 1,
                  "column": 8
                },
                "end": {
                  "line": 1,
                  "column": 19
                },
                "identifierName": "dispatchers"
              },
              "name": "dispatchers"
            }
          }
        ],
        "source": {
          "type": "StringLiteral",
          "start": 26,
          "end": 41,
          "loc": {
            "start": {
              "line": 1,
              "column": 26
            },
            "end": {
              "line": 1,
              "column": 41
            }
          },
          "extra": {
            "rawValue": "./dispatchers",
            "raw": "'./dispatchers'"
          },
          "value": "./dispatchers"
        }
      },
      {
        "type": "ImportDeclaration",
        "start": 43,
        "end": 81,
        "loc": {
          "start": {
            "line": 2,
            "column": 0
          },
          "end": {
            "line": 2,
            "column": 38
          }
        },
        "specifiers": [
          {
            "type": "ImportSpecifier",
            "start": 51,
            "end": 60,
            "loc": {
              "start": {
                "line": 2,
                "column": 8
              },
              "end": {
                "line": 2,
                "column": 17
              }
            },
            "imported": {
              "type": "Identifier",
              "start": 51,
              "end": 60,
              "loc": {
                "start": {
                  "line": 2,
                  "column": 8
                },
                "end": {
                  "line": 2,
                  "column": 17
                },
                "identifierName": "selectors"
              },
              "name": "selectors"
            },
            "local": {
              "type": "Identifier",
              "start": 51,
              "end": 60,
              "loc": {
                "start": {
                  "line": 2,
                  "column": 8
                },
                "end": {
                  "line": 2,
                  "column": 17
                },
                "identifierName": "selectors"
              },
              "name": "selectors"
            }
          }
        ],
        "source": {
          "type": "StringLiteral",
          "start": 67,
          "end": 80,
          "loc": {
            "start": {
              "line": 2,
              "column": 24
            },
            "end": {
              "line": 2,
              "column": 37
            }
          },
          "extra": {
            "rawValue": "./selectors",
            "raw": "'./selectors'"
          },
          "value": "./selectors"
        }
      },
      {
        "type": "ExportDefaultDeclaration",
        "start": 83,
        "end": 126,
        "loc": {
          "start": {
            "line": 4,
            "column": 0
          },
          "end": {
            "line": 7,
            "column": 2
          }
        },
        "declaration": {
          "type": "ObjectExpression",
          "start": 97,
          "end": 125,
          "loc": {
            "start": {
              "line": 4,
              "column": 14
            },
            "end": {
              "line": 7,
              "column": 1
            }
          },
          "properties": [
            {
              "type": "ObjectProperty",
              "start": 100,
              "end": 111,
              "loc": {
                "start": {
                  "line": 5,
                  "column": 1
                },
                "end": {
                  "line": 5,
                  "column": 12
                }
              },
              "method": false,
              "shorthand": true,
              "computed": false,
              "key": {
                "type": "Identifier",
                "start": 100,
                "end": 111,
                "loc": {
                  "start": {
                    "line": 5,
                    "column": 1
                  },
                  "end": {
                    "line": 5,
                    "column": 12
                  },
                  "identifierName": "dispatchers"
                },
                "name": "dispatchers"
              },
              "value": {
                "type": "Identifier",
                "start": 100,
                "end": 111,
                "loc": {
                  "start": {
                    "line": 5,
                    "column": 1
                  },
                  "end": {
                    "line": 5,
                    "column": 12
                  },
                  "identifierName": "dispatchers"
                },
                "name": "dispatchers"
              },
              "extra": {
                "shorthand": true
              }
            },
            {
              "type": "ObjectProperty",
              "start": 114,
              "end": 123,
              "loc": {
                "start": {
                  "line": 6,
                  "column": 1
                },
                "end": {
                  "line": 6,
                  "column": 10
                }
              },
              "method": false,
              "shorthand": true,
              "computed": false,
              "key": {
                "type": "Identifier",
                "start": 114,
                "end": 123,
                "loc": {
                  "start": {
                    "line": 6,
                    "column": 1
                  },
                  "end": {
                    "line": 6,
                    "column": 10
                  },
                  "identifierName": "selectors"
                },
                "name": "selectors"
              },
              "value": {
                "type": "Identifier",
                "start": 114,
                "end": 123,
                "loc": {
                  "start": {
                    "line": 6,
                    "column": 1
                  },
                  "end": {
                    "line": 6,
                    "column": 10
                  },
                  "identifierName": "selectors"
                },
                "name": "selectors"
              },
              "extra": {
                "shorthand": true
              }
            }
          ],
          "leadingComments": [],
          "trailingComments": []
        }
      }
    ],
    "directives": []
  },
  "comments": [],
  "tokens": [
    {
      "type": {
        "label": "import",
        "keyword": "import",
        "beforeExpr": false,
        "startsExpr": true,
        "rightAssociative": false,
        "isLoop": false,
        "isAssign": false,
        "prefix": false,
        "postfix": false,
        "binop": null,
        "updateContext": null
      },
      "value": "import",
      "start": 0,
      "end": 6,
      "loc": {
        "start": {
          "line": 1,
          "column": 0
        },
        "end": {
          "line": 1,
          "column": 6
        }
      }
    },
    {
      "type": {
        "label": "{",
        "beforeExpr": true,
        "startsExpr": true,
        "rightAssociative": false,
        "isLoop": false,
        "isAssign": false,
        "prefix": false,
        "postfix": false,
        "binop": null
      },
      "start": 7,
      "end": 8,
      "loc": {
        "start": {
          "line": 1,
          "column": 7
        },
        "end": {
          "line": 1,
          "column": 8
        }
      }
    },
    {
      "type": {
        "label": "name",
        "beforeExpr": false,
        "startsExpr": true,
        "rightAssociative": false,
        "isLoop": false,
        "isAssign": false,
        "prefix": false,
        "postfix": false,
        "binop": null
      },
      "value": "dispatchers",
      "start": 8,
      "end": 19,
      "loc": {
        "start": {
          "line": 1,
          "column": 8
        },
        "end": {
          "line": 1,
          "column": 19
        }
      }
    },
    {
      "type": {
        "label": "}",
        "beforeExpr": false,
        "startsExpr": false,
        "rightAssociative": false,
        "isLoop": false,
        "isAssign": false,
        "prefix": false,
        "postfix": false,
        "binop": null
      },
      "start": 19,
      "end": 20,
      "loc": {
        "start": {
          "line": 1,
          "column": 19
        },
        "end": {
          "line": 1,
          "column": 20
        }
      }
    },
    {
      "type": {
        "label": "name",
        "beforeExpr": false,
        "startsExpr": true,
        "rightAssociative": false,
        "isLoop": false,
        "isAssign": false,
        "prefix": false,
        "postfix": false,
        "binop": null
      },
      "value": "from",
      "start": 21,
      "end": 25,
      "loc": {
        "start": {
          "line": 1,
          "column": 21
        },
        "end": {
          "line": 1,
          "column": 25
        }
      }
    },
    {
      "type": {
        "label": "string",
        "beforeExpr": false,
        "startsExpr": true,
        "rightAssociative": false,
        "isLoop": false,
        "isAssign": false,
        "prefix": false,
        "postfix": false,
        "binop": null,
        "updateContext": null
      },
      "value": "./dispatchers",
      "start": 26,
      "end": 41,
      "loc": {
        "start": {
          "line": 1,
          "column": 26
        },
        "end": {
          "line": 1,
          "column": 41
        }
      }
    },
    {
      "type": {
        "label": ";",
        "beforeExpr": true,
        "startsExpr": false,
        "rightAssociative": false,
        "isLoop": false,
        "isAssign": false,
        "prefix": false,
        "postfix": false,
        "binop": null,
        "updateContext": null
      },
      "start": 41,
      "end": 42,
      "loc": {
        "start": {
          "line": 1,
          "column": 41
        },
        "end": {
          "line": 1,
          "column": 42
        }
      }
    },
    {
      "type": {
        "label": "import",
        "keyword": "import",
        "beforeExpr": false,
        "startsExpr": true,
        "rightAssociative": false,
        "isLoop": false,
        "isAssign": false,
        "prefix": false,
        "postfix": false,
        "binop": null,
        "updateContext": null
      },
      "value": "import",
      "start": 43,
      "end": 49,
      "loc": {
        "start": {
          "line": 2,
          "column": 0
        },
        "end": {
          "line": 2,
          "column": 6
        }
      }
    },
    {
      "type": {
        "label": "{",
        "beforeExpr": true,
        "startsExpr": true,
        "rightAssociative": false,
        "isLoop": false,
        "isAssign": false,
        "prefix": false,
        "postfix": false,
        "binop": null
      },
      "start": 50,
      "end": 51,
      "loc": {
        "start": {
          "line": 2,
          "column": 7
        },
        "end": {
          "line": 2,
          "column": 8
        }
      }
    },
    {
      "type": {
        "label": "name",
        "beforeExpr": false,
        "startsExpr": true,
        "rightAssociative": false,
        "isLoop": false,
        "isAssign": false,
        "prefix": false,
        "postfix": false,
        "binop": null
      },
      "value": "selectors",
      "start": 51,
      "end": 60,
      "loc": {
        "start": {
          "line": 2,
          "column": 8
        },
        "end": {
          "line": 2,
          "column": 17
        }
      }
    },
    {
      "type": {
        "label": "}",
        "beforeExpr": false,
        "startsExpr": false,
        "rightAssociative": false,
        "isLoop": false,
        "isAssign": false,
        "prefix": false,
        "postfix": false,
        "binop": null
      },
      "start": 60,
      "end": 61,
      "loc": {
        "start": {
          "line": 2,
          "column": 17
        },
        "end": {
          "line": 2,
          "column": 18
        }
      }
    },
    {
      "type": {
        "label": "name",
        "beforeExpr": false,
        "startsExpr": true,
        "rightAssociative": false,
        "isLoop": false,
        "isAssign": false,
        "prefix": false,
        "postfix": false,
        "binop": null
      },
      "value": "from",
      "start": 62,
      "end": 66,
      "loc": {
        "start": {
          "line": 2,
          "column": 19
        },
        "end": {
          "line": 2,
          "column": 23
        }
      }
    },
    {
      "type": {
        "label": "string",
        "beforeExpr": false,
        "startsExpr": true,
        "rightAssociative": false,
        "isLoop": false,
        "isAssign": false,
        "prefix": false,
        "postfix": false,
        "binop": null,
        "updateContext": null
      },
      "value": "./selectors",
      "start": 67,
      "end": 80,
      "loc": {
        "start": {
          "line": 2,
          "column": 24
        },
        "end": {
          "line": 2,
          "column": 37
        }
      }
    },
    {
      "type": {
        "label": ";",
        "beforeExpr": true,
        "startsExpr": false,
        "rightAssociative": false,
        "isLoop": false,
        "isAssign": false,
        "prefix": false,
        "postfix": false,
        "binop": null,
        "updateContext": null
      },
      "start": 80,
      "end": 81,
      "loc": {
        "start": {
          "line": 2,
          "column": 37
        },
        "end": {
          "line": 2,
          "column": 38
        }
      }
    },
    {
      "type": {
        "label": "export",
        "keyword": "export",
        "beforeExpr": false,
        "startsExpr": false,
        "rightAssociative": false,
        "isLoop": false,
        "isAssign": false,
        "prefix": false,
        "postfix": false,
        "binop": null,
        "updateContext": null
      },
      "value": "export",
      "start": 83,
      "end": 89,
      "loc": {
        "start": {
          "line": 4,
          "column": 0
        },
        "end": {
          "line": 4,
          "column": 6
        }
      }
    },
    {
      "type": {
        "label": "default",
        "keyword": "default",
        "beforeExpr": true,
        "startsExpr": false,
        "rightAssociative": false,
        "isLoop": false,
        "isAssign": false,
        "prefix": false,
        "postfix": false,
        "binop": null,
        "updateContext": null
      },
      "value": "default",
      "start": 90,
      "end": 97,
      "loc": {
        "start": {
          "line": 4,
          "column": 7
        },
        "end": {
          "line": 4,
          "column": 14
        }
      }
    },
    {
      "type": {
        "label": "{",
        "beforeExpr": true,
        "startsExpr": true,
        "rightAssociative": false,
        "isLoop": false,
        "isAssign": false,
        "prefix": false,
        "postfix": false,
        "binop": null
      },
      "start": 97,
      "end": 98,
      "loc": {
        "start": {
          "line": 4,
          "column": 14
        },
        "end": {
          "line": 4,
          "column": 15
        }
      }
    },
    {
      "type": {
        "label": "name",
        "beforeExpr": false,
        "startsExpr": true,
        "rightAssociative": false,
        "isLoop": false,
        "isAssign": false,
        "prefix": false,
        "postfix": false,
        "binop": null
      },
      "value": "dispatchers",
      "start": 100,
      "end": 111,
      "loc": {
        "start": {
          "line": 5,
          "column": 1
        },
        "end": {
          "line": 5,
          "column": 12
        }
      }
    },
    {
      "type": {
        "label": ",",
        "beforeExpr": true,
        "startsExpr": false,
        "rightAssociative": false,
        "isLoop": false,
        "isAssign": false,
        "prefix": false,
        "postfix": false,
        "binop": null,
        "updateContext": null
      },
      "start": 111,
      "end": 112,
      "loc": {
        "start": {
          "line": 5,
          "column": 12
        },
        "end": {
          "line": 5,
          "column": 13
        }
      }
    },
    {
      "type": {
        "label": "name",
        "beforeExpr": false,
        "startsExpr": true,
        "rightAssociative": false,
        "isLoop": false,
        "isAssign": false,
        "prefix": false,
        "postfix": false,
        "binop": null
      },
      "value": "selectors",
      "start": 114,
      "end": 123,
      "loc": {
        "start": {
          "line": 6,
          "column": 1
        },
        "end": {
          "line": 6,
          "column": 10
        }
      }
    },
    {
      "type": {
        "label": "}",
        "beforeExpr": false,
        "startsExpr": false,
        "rightAssociative": false,
        "isLoop": false,
        "isAssign": false,
        "prefix": false,
        "postfix": false,
        "binop": null
      },
      "start": 124,
      "end": 125,
      "loc": {
        "start": {
          "line": 7,
          "column": 0
        },
        "end": {
          "line": 7,
          "column": 1
        }
      }
    },
    {
      "type": {
        "label": ";",
        "beforeExpr": true,
        "startsExpr": false,
        "rightAssociative": false,
        "isLoop": false,
        "isAssign": false,
        "prefix": false,
        "postfix": false,
        "binop": null,
        "updateContext": null
      },
      "start": 125,
      "end": 126,
      "loc": {
        "start": {
          "line": 7,
          "column": 1
        },
        "end": {
          "line": 7,
          "column": 2
        }
      }
    },
    {
      "type": {
        "label": "eof",
        "beforeExpr": false,
        "startsExpr": false,
        "rightAssociative": false,
        "isLoop": false,
        "isAssign": false,
        "prefix": false,
        "postfix": false,
        "binop": null,
        "updateContext": null
      },
      "start": 126,
      "end": 126,
      "loc": {
        "start": {
          "line": 7,
          "column": 2
        },
        "end": {
          "line": 7,
          "column": 2
        }
      }
    }
  ]
}