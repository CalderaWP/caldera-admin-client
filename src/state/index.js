import {
	prepareSettings,
	settingsProp
} from './prepareSettings';
import {
	resolvers
} from './resolvers';
import {
	formTemplateReducer
} from './templates-store';

/**
 * Export all state-related functions
 */
export default {
	prepareSettings,
	settingsProp,
	resolvers,
	formTemplateReducer
};