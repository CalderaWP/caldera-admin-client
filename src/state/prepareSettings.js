import {GENERAL_SETTINGS} from '../components/Settings/GeneralSettings/generalSettingsType';
import {PRO_SETTINGS} from '../components/Settings/ProSettings/proSettingsType';
import {prepareProSettings} from '../components/Settings/ProSettings/prepareProSettings';
import {prepareGeneralSettings} from '../components/Settings/GeneralSettings/prepareGeneralSettings';

/**
 * Get a property from settings object
 *
 * @param {Object} settings Settings to search in
 * @param {String} prop Name of enumerable key in object to find
 * @param {String|Number|Array} defaultValue Fallback value
 * @return {String|Number|Array}
 */
export const settingsProp = (settings,prop,defaultValue) =>{
	return settings.hasOwnProperty(prop) ? settings[prop] : defaultValue;
};

/**
 * Prepare Caldera Forms settings
 * @param {Object} settings Settings to prepare
 *
 * @return {{privacySettings: {}, [p: string]: {}}}
 */
export const prepareSettings = (settings) => {
	return {
		[GENERAL_SETTINGS]: prepareGeneralSettings(
			settingsProp(settings,'styleIncludes',{}),
			settingsProp(settings,'otherSettings',{}),
		),
		[PRO_SETTINGS]: prepareProSettings(
			settingsProp(settings,PRO_SETTINGS,{})
		),
		privacySettings : {

		}
	};

};