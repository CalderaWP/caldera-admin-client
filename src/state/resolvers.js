import apiClients from '../apiClients';

const {
	formsAdminApiClient,
	PrivacySettingsClient,
	GeneralSettingClient,
	ProLocalSettingClient,
	entriesClient
} = apiClients;

/**
 * All @wordpress/data resolvers for this app
 * 
 * @type {{getForm: (function(*=): *), getForms: (function(*=): *), getFormPreview: (function(*=): *), getFormPrivacySettings: (function(*=): MediaTrackSettings), getStyleIncludes: (function(): ({grid: boolean, form: boolean, alert: boolean}|settings.styleIncludes|{grid, form, alert})), getOtherSettings: (function(): function(): ({cdnEnable: boolean}|settings.otherSettings|{cdnEnable})), getCfProFormSetting(): *, getCfProSettings: (function(): MediaTrackSettings), getPageOfEntries: (function(*=): *)}}
 */
export const resolvers = {
	getForm: (formId) => formsAdminApiClient.getForm(formId),
	getForms: (page = 1) => formsAdminApiClient.getForms(page),
	getFormPreview: (formId) => formsAdminApiClient.getFormPreview(formId),
	getFormPrivacySettings: (formId) => PrivacySettingsClient.getSettings(formId),
	getStyleIncludes: async () => {
		const settings = await GeneralSettingClient.getSettings();
		return settings.styleIncludes;
	},
	getOtherSettings: () => async () => {
		const settings = await GeneralSettingClient.getSettings();
		return settings.otherSettings;
	},
	getCfProFormSetting() {
		return {};
	},
	getCfProSettings: () => ProLocalSettingClient.getSettings(),
	getPageOfEntries: (page = 1) => entriesClient.getEntries(page),

};