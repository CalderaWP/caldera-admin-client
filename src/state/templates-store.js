import {state} from "@caldera-labs/state/lib/module";
import {cfAdmin} from "../apiClients";

const {CALDERA_FORMS_STORE_SLUG} = state;


/**
 * Selector to get all form templates
 *
 * @param state
 * @return {shim|*|{starter_contact_form, variable_price_example, registration, simple_booking_form_example}}
 */
export const getFormTemplates = (state) => {
	return state.templates;
};

/**
 * Slug for form templates store
 * @type {string}
 */
export const CALDERA_FORMS_TEMPLATE_STORE_SLUG =  `${CALDERA_FORMS_STORE_SLUG}/TEMPLATES`;

/**
 * Name of action indicating that form templates should be set in store
 * @type {string}
 */
export const SET_FORM_TEMPLATES = `${CALDERA_FORMS_TEMPLATE_STORE_SLUG}/SET_FORM_TEMPLATES`;

/**
 * Action to set form templates in store
 * @param {Object} templates
 * @return {{type: string, templates: *}}
 */
export const setFormTemplates = (templates) => {
	return {
		type: SET_FORM_TEMPLATES,
		templates
	};
};


/**
 * Default value for form templates store
 */
export const DEFAULT_TEMPLATES  = cfAdmin.templates;


/**
 * Reducer for form templates
 *
 * @param {Object} state
 * @param {Object} action
 * @return {*}
 */
export const formTemplateReducer = ( state = DEFAULT_TEMPLATES, action ) => {
	switch( action.type ){
		case SET_FORM_TEMPLATES:
			return {
				...state,
				templates: action.templates
			};
		default:
			return state;
	}
};

export default {
	formTemplateReducer,
	DEFAULT_TEMPLATES,
	actions: setFormTemplates,
	selectors: getFormTemplates
}