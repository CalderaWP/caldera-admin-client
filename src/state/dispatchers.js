import {store} from '@caldera-labs/state'

/**
 * Dispatch action functions for use with mapDispatchToProps
 * @param dispatch
 * @return {{setForm: setForm, setForms: setForms, newForm: newForm}}
 */
export const dispatchers = (dispatch) => {
	return {
		setForm: (form) => {
			dispatch(
				store.actions.setForm(form)
			)
		},
		setForms: (forms) => {
			dispatch(
				store.actions.setForms(forms)
			)
		},
		newForm: () => {
			dispatch(
				store.actions.newForm()
			)
		}
	};
};