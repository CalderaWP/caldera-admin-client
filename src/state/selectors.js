import {store, state} from '@caldera-labs/state';

/**
 * Selectors from state to use with mapStateToProps
 * @param currentState
 * @return {{forms: *, getForms: (function(): *), getForm: (function(*=): *)}}
 */
export const selectors = (currentState) => {
	const formState = currentState[state.CALDERA_FORMS_STORE_SLUG];
	const privacyState = currentState[state.CALDERA_FORMS_PRIVACY_STORE_SLUG];
	const cfProState = currentState[state.CALDERA_FORMS_PRO_LOCAL_SETTINGS_SLUG];
	const settingsState = currentState[state.CALDERA_FORMS_SETTINGS_SLUG];
	const forms = store.selectors.getForms(formState);
	return {
		forms: Array.isArray(forms) ? {} : forms,
		getForms: () => store.selectors.getForms(formState),
		getForm: (formId) => store.selectors.getForm(formState, formId),
		getFormPreview: (formId) => store.selectors.getFormPreview(formState, formId),
		getFormPrivacySettings: (formId) => store.selectors.getFormPrivacySettings(formId, privacyState),
		getCfProSettings: () => store.selectors.getCfProSettings(cfProState),
		proSettings: store.selectors.getCfProSettings(cfProState),
		getCfProFormSetting: (formId) => store.selectors.getCfProFormSetting(formId,cfProState),
		getStyleIncludes: () => store.selectors.getStyleIncludes(settingsState),
		getOtherSettings: () => store.selectors.getOtherSettings(settingsState)
	};
};