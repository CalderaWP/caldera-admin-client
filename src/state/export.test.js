import adminState from './index';

const {dispatchers, selectors} = adminState;
import {dispatch} from '@wordpress/data'
import {store, state} from '@caldera-labs/state'

describe('admin state export', () => {
	it('exports dispatchers', () => {
		expect(typeof dispatchers).toBe('function');
	});
	it('exports selectors', () => {
		expect(typeof selectors).toBe('function');
	});

	describe('dispatchers', () => {
		it('exports setForm action ', () => {
			expect(typeof dispatchers(dispatch).setForm).toBe('function');
		});
		it('exports setForms action ', () => {
			expect(typeof dispatchers(dispatch).setForms).toBe('function');
		});
		it('exports newForm action ', () => {
			expect(typeof dispatchers(dispatch).newForm).toBe('function');
		});
		it('exports setEditForm action ', () => {
			expect(typeof dispatchers(dispatch).setEditForm).toBe('function');
		});
		it('exports unsetEditForm action ', () => {
			expect(typeof dispatchers(dispatch).unsetEditForm).toBe('function');
		});
		it('exports addFormPreview action ', () => {
			expect(typeof dispatchers(dispatch).addFormPreview).toBe('function');
		});
		it('exports setFormPrivacyForm action ', () => {
			expect(typeof dispatchers(dispatch).setFormPrivacyForm).toBe('function');
		});
		it('exports updateStyleIncludes action ', () => {
			expect(typeof dispatchers(dispatch).updateStyleIncludes).toBe('function');
		});
		it('exports updateOtherSettings action ', () => {
			expect(typeof dispatchers(dispatch).updateOtherSettings).toBe('function');
		});
		it('exports updateCfProFormSetting action ', () => {
			expect(typeof dispatchers(dispatch).updateCfProFormSetting).toBe('function');
		});
		it('exports updateCfProSettings action ', () => {
			expect(typeof dispatchers(dispatch).updateCfProSettings).toBe('function');
		});
	});

	describe('selectors', () => {
		const forms = [
			{ID: '1'}
		];
		const mockState = {
			[state.CALDERA_FORMS_STORE_SLUG]: {
				forms
			},
			[state.CALDERA_FORMS_PRIVACY_STORE_SLUG]: {
				forms
			},
			[state.CALDERA_FORMS_PRO_LOCAL_SETTINGS_SLUG]: {
				settings: {
					apiKeys:{}
				},
				forms

			},
			[state.CALDERA_FORMS_SETTINGS_SLUG]: {
				styleIncludes:{},
				other:{}
			},

		};
		it('exports forms object ', () => {
			expect(typeof selectors(mockState).forms).toBe('object');
		});

		it('can use forms object ', () => {
			expect(selectors(mockState).forms).toEqual({
				1: forms[0]
			});
		});

		it('exports getForms selectors ', () => {
			expect(typeof selectors(mockState).getForms).toBe('function');
		});
		it('exports getForm selectors ', () => {
			expect(typeof selectors(mockState).getForm).toBe('function');
		});

		it('exports getFormPreview selectors ', () => {
			expect(typeof selectors(mockState).getFormPreview).toBe('function');
		});
		it('exports getFormPrivacySettings selectors ', () => {
			expect(typeof selectors(mockState).getFormPrivacySettings).toBe('function');
		});
		it('exports getFormPreview selectors ', () => {
			expect(typeof selectors(mockState).getFormPreview).toBe('function');
		});
		it('exports getCfProSettings selectors ', () => {
			expect(typeof selectors(mockState).getCfProSettings).toBe('function');
		});
		it('exports getCfProFormSetting selectors ', () => {
			expect(typeof selectors(mockState).getCfProFormSetting).toBe('function');
		});
		it('exports getStyleIncludes selectors ', () => {
			expect(typeof selectors(mockState).getStyleIncludes).toBe('function');
		});
		it('exports getOtherSettings selectors ', () => {
			expect(typeof selectors(mockState).getOtherSettings).toBe('function');
		});
		it('exports proSettings ', () => {
			expect(typeof selectors(mockState).proSettings).toBe('object');
		});

	});
});