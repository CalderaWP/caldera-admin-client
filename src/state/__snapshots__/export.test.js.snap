// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`export has the right types and keys 1`] = `"{\\"prepareSettings\\":\\"function\\",\\"settingsProp\\":\\"function\\",\\"resolvers\\":\\"object\\",\\"formTemplateReducer\\":\\"function\\"}"`;
